railties/lib/rails/generators/rails/app/templates/Dockerfile.tt

```diff
@@ -32,19 +32,19 @@ RUN curl -sL https://github.com/nodenv/node-build/archive/master.tar.gz | tar xz
 
 <% end -%>
 # Install application gems
-COPY Gemfile Gemfile.lock ./
+COPY --link Gemfile Gemfile.lock ./
```

Comment:
Is docker really nor smart enough to notice we're copying two files?

Reply:
It's more because `--link` can't always be used, so you have to opt-in to it. It'll definitely be much more beneficial for the `COPY --link . .` command, but this will still have a minor benefit. The documentation recommends always using it when possible https://docs.docker.com/engine/reference/builder/#copy---link.